chanbackup/pubsub.go

```diff
@@ -155,12 +155,44 @@ func (s *SubSwapper) Stop() error {
 }
 
 // updateBackupFile updates the backup file in place given the current state of
-// the SubSwapper.
-func (s *SubSwapper) updateBackupFile() error {
+// the SubSwapper. We accept the set of channels that were closed between this
+// update and the last to make sure we leave them out of our backup set union.
+func (s *SubSwapper) updateBackupFile(closedChans ...wire.OutPoint) error {
+	// Before we pack the new set of SCBs, we'll first decode what we
+	// already have on-disk, to make sure we can decode it (proper seed)
+	// and that we're able to combine it with our new data.
+	diskMulti, err := s.Swapper.ExtractMulti(s.keyRing)
+	if err != nil {
+		return fmt.Errorf("unable to extract on disk encrypted "+
+			"SCB: %v", err)
+	}
+
+	// Now that we have channels stored on-disk, we'll create a new set of
+	// the combined old and new channels to make sure we retain what's
+	// already on-disk.
+	//
+	// NOTE: The ordering of this operations means that our in-memory
+	// structure will replace what we read from disk.
+	combinedBackup := make(map[wire.OutPoint]Single)
+	if diskMulti != nil {
+		for _, diskChannel := range diskMulti.StaticBackups {
+			chanPoint := diskChannel.FundingOutpoint
+			combinedBackup[chanPoint] = diskChannel
+		}
+	}
+	for _, memChannel := range s.backupState {
+		combinedBackup[memChannel.FundingOutpoint] = memChannel
```

Comment:
perhaps we should check that if we find a collision we are setting it with the same value? just as a sanity check

Reply:
What would the behavior be if the value is different? I think it's better to _replace_ the existing, as the new one might have a new field, or be using a new SCB version. If we want to replace, then this logic implements that as is.

Reply:
It's also possible to just leave the duplicate there too.

Reply:
that's true, i was thinking even a warning would be nice to see. you're right though the newer one is probably more correct

Reply:
log warning added

Reply:
Realize now, that the way it is, it'll always print that log statement in the typical case...